caddyhttp/httpserver/replacer.go

```diff
@@ -37,8 +37,8 @@ type Replacer interface {
 // they will be used to overwrite other replacements
 // if there is a name conflict.
 type replacer struct {
-	replacements       map[string]string
-	customReplacements map[string]string
+	replacements       map[string]func() string
+	customReplacements map[string]func() string
```

Comment:
Ahhh... this makes sense. Does using functions to defer evaluation of these values have anything to do with #916 (where the `Host: {host}` was 'cached' between different hostnames)?